klusterletconfig/v1alpha1/types.go

```diff
@@ -124,6 +139,88 @@ type Registries struct {
 	Source string `json:"source"`
 }
 
+// KubeAPIServerConfig specifies the custom configuration for the Hub kube API server
+type KubeAPIServerConfig struct {
+	// URL is the endpoint of the hub Kube API server.
+	// If not present, the .status.apiServerURL of Infrastructure/cluster will be used as the default value.
+	// e.g. `oc get infrastructure cluster -o jsonpath='{.status.apiServerURL}'`
+	// +optional
+	URL string `json:"url,omitempty"`
+
+	// ServerVerificationStrategy is the strategy used for verifying the server certification
+	// +kubebuilder:default=default
+	// +kubebuilder:validation:Enum=use-system-truststore;use-auto-detected-truststore;use-custom-ca-bundles;default
+	// +optional
+	ServerVerificationStrategy ServerVerificationStrategy `json:"serverVerificationStrategy,omitempty"`
+
+	// TrustedCABundles refers to a collection of user-provided CA bundles used for verifying the server
+	// certificate of the hub Kubernetes API
+	// If the ServerVerificationStrategy is set to "use-system-truststore", this field will be ignored.
+	// Otherwise, the CA certificates from the configured bundles will be appended to the klusterlet CA bundle.
+	// +listType:=map
+	// +listMapKey:=name
+	// +optional
+	TrustedCABundles []CABundle `json:"trustedCABundles,omitempty"`
+
+	// ProxyURL is the URL to the proxy to be used for all requests made by client
+	// If an HTTPS proxy server is configured, you may also need to add the necessary CA certificates to
+	// TrustedCABundles.
+	// +optional
+	ProxyURL string `json:"proxyURL,omitempty"`
+}
+
+// CABundle is a user-provided CA bundle
+type CABundle struct {
+	// Name is the identifier used to reference the CA bundle
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name,omitempty"`
+
+	// CABundleData is base64 encoded certificate data
+	// +optional
+	CABundleData []byte `json:"caBundleData,omitempty"`
+
+	// CABundle refers to a ConfigMap containing the user-provided CA bundle
+	// The key of the CA data must be "ca-bundle.crt".
+	// This field will be ignored if the CABundleData field is also present.
+	// +optional
+	CABundle *ConfigMapReference `json:"caBundle,omitempty"`
+}
+
+// ServerVerificationStrategy represents the strategy used for the server certificate varification
+type ServerVerificationStrategy string
+
+const (
+	// ServerVerificationStrategyUseSystemTruststore is the strategy that utilizes CA certificates in the system
+	// truststore of the Operating System to validate the server certificate.
+	ServerVerificationStrategyUseSystemTruststore ServerVerificationStrategy = "use-system-truststore"
+
+	// ServerVerificationStrategyUseAutoDetectedTruststore is the strategy that automatically detects CA certificates
+	// for the hub Kube API server and uses them to validate the server certificate.
+	ServerVerificationStrategyUseAutoDetectedTruststore ServerVerificationStrategy = "use-auto-detected-truststore"
```

Comment:
I'd like to rename it to `UseAutoDetectedCABundle` because the auto-detected CA certificates will be shown as a CA bundle in the status.

Reply:
ok, so it will be`UseAutoDetectedCABundle` and `UseCustomCABundles`, should we change it to `UseAutoDetectedCABundles`(add an ending `s`)?